bundles/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudApiResponseException.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.icloud.internal;
+
+/**
+ *
+ * Exception for errors during calls of the iCloud API.
```

Comment:
Normally I avoid using checked exception. But this is a endless discussion in java space. If it is more common to use checked exception in openHAB for this types for errors (e.g. expected exceptions?!), I would change it. If it is more my personal opinion I would like to leave it as it is.

Reply:
Well, a `RuntimeException` is usually considered a programming error and is fatal. In this case we should expect API errors from time to time and must always handle these to avoid that the binding will stop working. So for this specific case, I would definitely recommend a checked exception.

I now double-checked our review checklist, and actually this is mentioned as a requirement: "Don't throw `RuntimeException` on expected errors" - for reference: https://github.com/openhab/openhab-addons/wiki/Review-Checklist.

Reply:
I'll changed it to checked exception.